src/lithopscloud/modules/cloud_functions/cloud_function.py

```diff
@@ -0,0 +1,147 @@
+import http.client
+import json
+import sys
+from typing import Dict, Any
+import requests
+from lithopscloud.modules.config_builder import ConfigBuilder, spinner
+from lithopscloud.modules.utils import get_option_from_list_alt, free_dialog, color_msg, Color
+
+CF_REGIONS = ['eu-de', 'eu-gb', 'us-south', 'us-east', 'au-syd', 'jp-tok']
+
+
+class CloudFunction(ConfigBuilder):
+
+    def __init__(self, base_config: Dict[str, Any]) -> None:
+        super().__init__(base_config)
+
+    def run(self, api_key=None) -> Dict[str, Any]:
+
+        is_cf_based_namespace = False  # set to true if namespace is cloud foundry based
+        print(color_msg("\n\nConfiguring IBM cloud functions:\n",color=Color.YELLOW))
+
+        choices = CF_REGIONS[:]
+
+        choices.insert(0, "Search across all regions. -*- Will take considerably longer to attain -*-")
+        selected_region = get_option_from_list_alt("Choose a region where your preferred namespace can be found,"
+                                                   " or created in", choices)['answer']
+
+        if 'Present all existing namespaces' in selected_region:
+            selected_region = ''  # mark region as not yet selected.
+
+        existing_namespaces = self.get_cloud_function_namespaces(selected_region)
+
+        msg = 'Please pick one of your cloud function namespaces '
+        msg = msg + f'in the {selected_region} region' if selected_region else msg
+        chosen_namespace = get_option_from_list_alt(msg,
+                                                    list(existing_namespaces.keys()),
+                                                    instance_to_create='namespace',
+                                                    default=self.base_config['ibm_cf']['namespace'])['answer']
+
+        if chosen_namespace == 'Create a new namespace':
+            if not selected_region:
+                selected_region = get_option_from_list_alt('Please choose the region you would like '
+                                                           'to create a namespace in :', CF_REGIONS)['answer']
+            namespace_id, chosen_namespace = self.create_cloud_function_namespaces(selected_region)
+            del self.base_config['ibm_cf']['api_key']
+
+        else:  # user would like to use an already existing namespace
+            selected_region = existing_namespaces[chosen_namespace]['region']
+
+            if existing_namespaces[chosen_namespace]['type'] == 'CF_based':
+                cf_api_key = input("Please provide your cloud foundry api_key from: "
+                                   "https://cloud.ibm.com/functions/namespace-settings ")
```

Comment:
Can't we get it programmatically?

Reply:
I don't believe we can get the apikey of cloud foundry based namespaces, the same way we can't pull the iam api key of a user. there isn't a way to list it via the ibmcloud cli tool, as far as i'm aware.